.github/workflows/betabuild.yml

```diff
@@ -10,39 +10,73 @@ on: push
   build:
+    outputs:
+      artifact_name: ${{ steps.fname.outputs.result }}
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-    - uses: actions/checkout@v2
-      
-    - uses: actions/github-script@v2
-      id: fname
-      with:
-        result-encoding: string
-        script: |
-          const fs = require("fs");
-          let file = fs.readFileSync("./gradle.properties");
-          file = file.toString().split("\n").map(e => e.trim().startsWith("mod_version") ? `${e}-beta-${process.env.GITHUB_SHA.substring(0, 7)}` : e).join("\n");
-          fs.writeFileSync("./gradle.properties", file);
-          const mod_version = file.toString().split("\n").filter(e => e.trim().startsWith("mod_version"))[0].split("=")[1].trim();
-          const archives_base_name = file.toString().split("\n").filter(e => e.trim().startsWith("archives_base_name"))[0].split("=")[1].trim();
-          return `${archives_base_name}-${mod_version}`;
-      
-    - name: Set up JDK 17
-      uses: actions/setup-java@v2
-      with:
-       distribution: 'adopt'
-       java-version: '17'
-        
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-      
-    - name: Build with Gradle
-      run: ./gradlew build
-
-    - uses: actions/upload-artifact@v2
-      with:
-        name: ${{ steps.fname.outputs.result }}
-        path: ./dist/**/*
\ No newline at end of file
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - uses: actions/github-script@v2
+        id: fname
+        with:
+          result-encoding: string
+          script: |
+            const fs = require("fs");
+            let file = fs.readFileSync("./gradle.properties");
+            file = file.toString().split("\n").map(e => e.trim().startsWith("mod_version") ? `${e}-beta-${process.env.GITHUB_SHA.substring(0, 7)}` : e).join("\n");
+            fs.writeFileSync("./gradle.properties", file);
+            const mod_version = file.toString().split("\n").filter(e => e.trim().startsWith("mod_version"))[0].split("=")[1].trim();
+            const archives_base_name = file.toString().split("\n").filter(e => e.trim().startsWith("archives_base_name"))[0].split("=")[1].trim();
+            return `${archives_base_name}-${mod_version}`;
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Build with Gradle
+        run: ./gradlew build
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ steps.fname.outputs.result }}
+          path: ./dist/**/*
+
+  publish_npm:
```

Comment:
beta shouldn't publish

Reply:
Will implement as requested in about 8 hours when i get home, but i would urge you to reconsider.

npm installs @latest by default (and the CI file only tags release builds with @latest), so the only way for someone to get a beta build is if they explicitly ask for it by installing either `@beta` or `0.0.0-shortsha`.
Therefore i see no downside to having both. 

On the other hand, only having release versions on npm means that anyone trying a beta (or instructed to try a beta) has to learn to do it manually anyway, at which point there's no reason for them to use the package at all since doing it manually works for *any* version

---
In the test package both tags @latest and @beta point to a beta version, that was done manually to make it easier to compare the type issues and should not happen with the CI as it currently stands (note the `--tag beta` in the betabuild publish script).

Reply:
reverted as requested

Reply:
you do have a point... hmm...

Reply:
Restored per https://github.com/JsMacros/JsMacros/pull/62#issuecomment-1319527005